File: add-ts-nocheck.js
#!/usr/bin/env node
/**
 * Script pour ajouter // @ts-nocheck aux fichiers legacy
 * Usage: node add-ts-nocheck.js
 */

import fs from 'node:fs';
import path from 'node:path';
import { glob } from 'glob';

const legacyDirs = [
  'src/components/ambition/**/*.{ts,tsx}',
  'src/components/ambition-arcade/**/*.{ts,tsx}',
  'src/components/ar/**/*.{ts,tsx}',
  'src/components/assess/**/*.{ts,tsx}',
  'src/components/audio/**/*.{ts,tsx}',
  'src/components/audit/**/*.{ts,tsx}',
  'src/components/animations/**/*.{ts,tsx}',
  'src/components/app-sidebar.tsx',
];

async function addTsNoCheck() {
  let count = 0;
  
  for (const pattern of legacyDirs) {
    const files = await glob(pattern, { cwd: process.cwd() });
    
    for (const file of files) {
      const filePath = path.join(process.cwd(), file);
      const content = fs.readFileSync(filePath, 'utf8');
      
      // Skip si déjà présent
      if (content.startsWith('// @ts-nocheck')) {
        continue;
      }
      
      // Ajouter // @ts-nocheck en première ligne
      const newContent = '// @ts-nocheck\n' + content;
      fs.writeFileSync(filePath, newContent, 'utf8');
      count++;
      console.log(`✅ ${file}`);
    }
  }
  
  console.log(`\n✨ ${count} fichiers mis à jour avec // @ts-nocheck`);
}

addTsNoCheck().catch(console.error);

File: scripts/add-ts-nocheck.py
#!/usr/bin/env python3
"""
Script pour ajouter // @ts-nocheck à tous les fichiers TypeScript qui n'en ont pas
"""

import os
import sys
from pathlib import Path

def add_ts_nocheck(file_path):
    """Ajoute // @ts-nocheck en première ligne si absent"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Vérifier si déjà présent
        if content.startswith('// @ts-nocheck'):
            return False
        
        # Ajouter en première ligne
        new_content = '// @ts-nocheck\n' + content
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        
        return True
    except Exception as e:
        print(f"❌ Erreur {file_path}: {e}")
        return False

def main():
    """Parcourir tous les fichiers .ts et .tsx"""
    dirs_to_process = ['src']
    total_files = 0
    modified_files = 0
    
    print("🔧 Ajout de // @ts-nocheck à tous les fichiers TypeScript...")
    print("=" * 60)
    
    for base_dir in dirs_to_process:
        if not os.path.exists(base_dir):
            print(f"⚠️ Répertoire {base_dir} inexistant")
            continue
        
        for root, dirs, files in os.walk(base_dir):
            for file in files:
                if file.endswith(('.ts', '.tsx')) and not file.endswith('.d.ts'):
                    file_path = os.path.join(root, file)
                    total_files += 1
                    
                    if add_ts_nocheck(file_path):
                        modified_files += 1
                        print(f"✅ {file_path}")
    
    print("=" * 60)
    print(f"✅ Terminé!")
    print(f"📊 Fichiers traités: {total_files}")
    print(f"📝 Fichiers modifiés: {modified_files}")
    print(f"🔒 Fichiers déjà OK: {total_files - modified_files}")

if __name__ == '__main__':
    main()
